### ISSUE TITLE
Debugging should switch focus to new stopping thread after exec 'continue' command

### ISSUE THREAD

**Issue by testforstephen** [microsoft/vscode#68955]
Here is the vscode insider version where i tried java multiple threads program. It still has similar switching problem as the issue #55862 when exec `continue` operation.

```
Commit: 709bf352b12de8284683bef94cb5984129bfc776
Date: 2019-02-08T14:53:26.555Z
Electron: 3.1.3
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17763
```

Below is the java program ThreadTest.java i used. Add two breakpoints at line 14 and line 27. When exec `next`, it doesn't auto switch thread focus, this is expected. But when exec `continue`, the thread focus always stay on one thread, not auto jump to other suspend threads.

 <details><summary> ThreadTest.java</summary>
<pre>
import java.util.concurrent.atomic.AtomicInteger;

public class ThreadTest {
    public static void main(String[] args) throws Exception {        
        Object object = new Object();
        AtomicInteger ai = new AtomicInteger();
        Thread thread0 = new Thread(() -> {
            for (;;) {
                System.out.println(String.format("Print %d in thread 0" , ai.incrementAndGet()));
                synchronized(object) {
                    object.notify();
                }
                try {
                    Thread.sleep(1000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                    break;
                }
            }
        });
        Thread thread1 = new Thread(() -> {
            for (;;) {
                try {
                    synchronized(object) {
                        object.wait();
                    }
                    System.out.println(String.format("Print %d in thread 1" , ai.incrementAndGet()));
                } catch (InterruptedException e) {
                    e.printStackTrace();
                    break;
                }
            }
        });

        thread0.start();
        thread1.start();
        thread0.join();
        thread1.join();
    }
}
</pre>
</details>
Below are the reproduce steps in vscode-insider:

1. Add two breakpoints at line 14 and line 27.
2. Start java debug, there are two breakpoints hit at line 14 (thread-0) and 27 (thread-1).
3. Select thread-0 to continue, line 14 will be hit again and the focus always stay on thread-0. The focus never switch to thread-1.  As comparison, try the same steps both in eclipse and intellij, click `continue` and it will switch back and forth between thread-0 and thread-1.  I think we should maintain a queue to store all suspend threads, every time when exec `continue`, pop one. And a new breakpoint event arrives, push it.

The focus switch problem is important because java supports Thread Level suspend, that means you can choose to `continue` one stopping thread, and keep other threads still on stopping state. Other languages, like nodejs, continue will resume the whole program, so it doesn't need care about the focus switch.

**Comment by isidorn** [comment]
Note to self: to change this the code here https://github.com/Microsoft/vscode/blob/966050a6531e5794c5ca45cd9c724d1b3d108cc7/src/vs/workbench/contrib/debug/electron-browser/debugService.ts#L492 
should call into debugService.setFocus to auto focus a possible thread.
However that should only be done if the continued action was done.

This might introduce some ugly code, as the session would have to remember if the last request it did was a continue request.
Bottom line: this is a feature request and we willl not do it this milestone.

**Comment by jkassis** [comment]
one command to "step forward in this gofunc", and another command to "step forward in any gofunc" would be nice.

**Comment by AdiL2608** [comment]
Unfortunatly I still experience this problem in a Django application. 
When the breakpoint is hit, the debugger would always focus on a different thread where there is no breakpoint.

It always follows this pattern in the call stack:

1st step of call stack: in __call__ method of CustomMiddleWare class
2nd step of call stack: in wrapper_func when calling func of custom_decorator
2rd step of call stack: inside of the called function which was called from custom_decorator, the focus of the debugger is here

I added the steps in the code below as comments

custommiddlewaremodule.py:

```
class CustomMiddleWare():
    def __init__(self, get_response):
        self.get_response = get_response

    def __call__(self, request):
        response = self.get_response(request) # 1st step in call stack
        return response

def custom_decorator_func():
    def decorator_func(func, *args, **kwargs):
        def wrapper_func(request, *args, **kwargs):
			# decoration logic
			return func(request, *args, **kwargs)  # 2nd step in call stack
        return wrapper_func
    return decorator_func

# 3rd step in call stack = called func which was passed to the custom_decorator  # FOCUS OF DEBUGGER IS HERE
```
	
settings.py:

```
MIDDLEWARE = [
    'custommiddlewaremodule.CustomMiddleWare',
	# rest of the middleware
	]
```
